staging/src/k8s.io/kubectl/pkg/cmd/apply/apply.go

```diff
@@ -93,7 +97,9 @@ type ApplyOptions struct {
 	All                     bool
 	Overwrite               bool
 	OpenAPIPatch            bool
-	PruneWhitelist          []string
+
+	// DEPRECATED: Not used
+	PruneWhitelist []string // TODO: Remove this in kubectl 1.29 or later
```

Comment:
Is this because of the flag/option refactor initiative? Why leave the field in the struct if the behaviour is in fact broken if the consumer doesn't make the change?

Reply:
I don't know why this field was not being used (this PR isn't what makes it unused, it was already unused).

I could just go ahead and remove it, but I was thinking the deprecation period would apply not only to command-line usage, but also to any exported members.

I'm OK with (and prefer) removing it altogether, if that is OK to do.  I guess all that could happen is someone tries to depend on the newer kubectl and their build would fail if they were setting this field for some reason. It's probably a better outcome anyway, because it wouldn't be doing what they think it does.

Reply:
I agree with removing it then. I don't see a purpose in retaining something that already isn't functional--it can only mislead people.